lib/mix/tasks/release.clean.ex

```diff
@@ -52,10 +53,9 @@ defmodule Mix.Tasks.Release.Clean do
 
   # Clean release build
   def do_cleanup(:build) do
-    cwd       = File.cwd!
     project   = Mix.Project.config |> Keyword.get(:app) |> Atom.to_string
     version   = Mix.Project.config |> Keyword.get(:version)
-    build     = Path.join([cwd, "_build", "prod"])
+    build     = Mix.Project.build_path
```

Comment:
In fact, this failure is result of cross-talking tests. When ran in isolation the test succeed.